src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,61 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int STARTING_CAPACITY = 10;
+    private K[] keys = (K[]) new Object[STARTING_CAPACITY];
+    private V[] values = (V[]) new Object[STARTING_CAPACITY];
+    private int size = 0;
+
     @Override
     public void put(K key, V value) {
+        int keyIndex = getKeyIndex(key);
+        if (keyIndex == -1) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            values[keyIndex] = value;
+        }
+        if (keys.length == size) {
+            increaseCapacity();
+        }
     }
 
     @Override
     public V get(K key) {
+        int keyIndex = getKeyIndex(key);
+        if (keyIndex != -1) {
+            return values[keyIndex];
+        }
         return null;
     }
 
     @Override
     public int size() {
+        return size;
+    }
+
+    //returns -1 if no such key was found
+    private int getKeyIndex(K obj) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] != null && keys[i].equals(obj)) {
+                return i;
+            }
+            if (keys[i] == null && obj == null) {
+                return i;
+            }
+        }
         return -1;
     }
+
+    //doubles arrays length
+    private void increaseCapacity() {
+        K[] newKeysArray = (K[]) new Object[keys.length * 2];
+        System.arraycopy(keys, 0, newKeysArray, 0, keys.length);
+        keys = newKeysArray;
+
+        V[] newValuesArray = (V[]) new Object[values.length * 2];
+        System.arraycopy(values, 0, newValuesArray, 0, values.length);
+        values = newValuesArray;
+    }
```

Comment:
```suggestion
    //doubles arrays length
    private void increaseCapacity() {
        K[] newKeysArray = (K[]) new Object[keys.length * 2];
        System.arraycopy(keys, 0, newKeysArray, 0, keys.length);
        keys = newKeysArray;

        V[] newValuesArray = (V[]) new Object[values.length * 2];
        System.arraycopy(values, 0, newValuesArray, 0, values.length);
        values = newValuesArray;
    }
```
we shouldn`t use this method, bacause we assume that the maximum number of elements in our Storage is 10

Reply:
Added it just in case. I think it doesnt't hurt to have it)